Added parameter to disallow calendar subscription via link

pboguslawski opened the pull request:
Added calendar application config parameter `allow_subscribe_link` to disallow calendar subscription via link (i.e. for internal Nextcloud setups). Use

```
occ config:app:set calendar allow_subscribe_link --value 'yes'
```

to allow (default if not set) and

```
occ config:app:set calendar allow_subscribe_link --value 'no'
```

to disallow calendar subscription via link.

Author-Change-Id: IB#1126265
Signed-off-by: Pawel Boguslawski <pawel.boguslawski@ib.pl>

pboguslawski committed:
Added parameter to disallow calendar subscription via link

Added calendar application config parameter `allow_subscribe_link` to disallow calendar
subscription via link (i.e. for internal Nextcloud setups). Use

```
occ config:app:set calendar allow_subscribe_link --value 'yes'
```

to allow (default if not set) and

```
occ config:app:set calendar allow_subscribe_link --value 'no'
```

to disallow calendar subscription via link.

Author-Change-Id: IB#1126265
Signed-off-by: Pawel Boguslawski <pawel.boguslawski@ib.pl>

tcitworld commented on apps/dav/lib/CalDAV/CalDavBackend.php:
```diff
@@ -2424,6 +2425,10 @@ public function getSubscriptionsForUser($principalUri) {
 	 */
 	public function createSubscription($principalUri, $uri, array $properties) {
+		// Subscription via link must be enabled.
+		if ($this->config->getAppValue('calendar', 'allow_subscribe_link', 'yes') === 'no') {
```
This should be directly disabled through not registering the plugin here: 
https://github.com/nextcloud/server/blob/754cef369b694c5a58f006ceafe7e9a943c7770f/apps/dav/lib/Server.php#L180

pboguslawski committed:
Moved to disabling through not registering the plugin

Related: https://github.com/nextcloud/server/pull/34387#discussion_r985250812
Fixes: 1e8bd2591146398e8d984cbdf73212670a1c8b46
Author-Change-Id: IB#1126265
Signed-off-by: Pawel Boguslawski <pawel.boguslawski@ib.pl>

tcitworld commented on apps/dav/lib/Server.php:
```diff
@@ -177,7 +178,9 @@ public function __construct(IRequest $request, string $baseUri) {
 			$this->server->addPlugin(\OC::$server->get(\OCA\DAV\CalDAV\Trashbin\Plugin::class));
 			$this->server->addPlugin(new \OCA\DAV\CalDAV\WebcalCaching\Plugin($request));
-			$this->server->addPlugin(new \Sabre\CalDAV\Subscriptions\Plugin());
+			if (\OC::$server->getConfig()->getAppValue('calendar', 'allow_subscribe_link', 'yes') === 'yes') {
```
```suggestion
			if (\OC::$server->getConfig()->getAppValue('dav', 'allow_subscribe_link', 'yes') === 'yes') {
```
Makes more sense in the `dav` app. Needs https://github.com/nextcloud/calendar/pull/4573 to be adjusted too.

tcitworld replied:
The calendar app is only a CalDAV client, settings in this app should only concern UI stuff. You can use subscriptions without the calendar app (for instance with DAVx5).

To make things clear from other dav stuff I suggest you rename `allow_subscribe_link` in `allow_calendar_subscriptions`.

pboguslawski replied:
Thanks for clarification. Param name [changed](https://github.com/nextcloud/server/pull/34387/commits/6128163b8a56a7c3f13d7a4f56beb74d6d7622eb) to `dav.allow_calendar_link_subscriptions` to distinguish it from other subscription methods.

pboguslawski committed:
Param renamed to dav.allow_calendar_link_subscriptions

Fixes: 1e8bd2591146398e8d984cbdf73212670a1c8b46
Related: https://github.com/nextcloud/server/pull/34387#discussion_r1008131871
Author-Change-Id: IB#1126265
Signed-off-by: Pawel Boguslawski <pawel.boguslawski@ib.pl>

tcitworld reviewed (approved):
Please squash your commits

pboguslawski commented:
> Please squash your commits

Don't see such option in github UI nor have spare time for such cleanup outside now. Please consider [squashing on merge](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/incorporating-changes-from-a-pull-request/about-pull-request-merges#squash-and-merge-your-commits) if you still see it valuable or just accept PR as is with all history included.

ChristophWurst commented:
We've tried to squash the PR for you but we lack the permission to push to your fork.

pboguslawski commented:
Deleted branch by accident which closed this PR - sorry. Please find squashed PR on
https://github.com/nextcloud/server/pull/34920
